src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests3.cs

```diff
@@ -1386,5 +1386,38 @@ public static (int?, int) Get(string name)
 ";
             CompileAndVerify(CreateCompilation(source, options: TestOptions.ReleaseExe).VerifyDiagnostics(), expectedOutput: "null");
         }
+
+        [Fact, WorkItem(38686, "https://github.com/dotnet/roslyn/issues/38686")]
+        public void TargetTypedSwitch_AnyTypedSwitchWithoutTargetType()
```

Comment:
>TargetTypedSwitch_AnyTypedSwitchWithoutTargetType [](start = 20, length = 49)

Please add a test for a scenario when similar "invalid" conversion is applied implicitly. For IOperation as well.

Reply:
Can you please give the specific scenario you'd like tested (i.e. the code to be compiled)?

---
In reply to: [326337342](https://github.com/dotnet/roslyn/pull/38750#discussion_r326337342) [](ancestors = 326337342)

Reply:
> Can you please give the specific scenario you'd like tested (i.e. the code to be compiled)?

I don't have a specific scenario at the moment. Perhaps assigning to a variable typed ```C``` instead of using a cast operator would do, I don't have that intimate knowledge of this area (target-typing + switch expressions) in the compiler. The title of the PR says: "Ensure target-typed constructs are bound during initial binding in an error scenario." It only changes and tests explicit cast scenario. I am assuming that target typing can occur with implicit conversions as well. We need to make sure that "target-typed constructs are bound during initial binding in an error scenario" for those cases too. At the very least we should have tests verifying that. I believe you added target-typing infrastructure into the compiler and added support for switch expressions, therefore, I think you have better chances to come up with right scenarios to test than anyone else on the team. It is quite possible we already have tests like these, then we need to add IOperation tree verification.

---
In reply to: [326408524](https://github.com/dotnet/roslyn/pull/38750#discussion_r326408524) [](ancestors = 326408524,326337342)

Reply:
We have assertions in the compiler to ensure target typing for many other error scenarios, and they have caught hundreds of places that needed to be (and were) modified.  This specific one slipped through.  We already have tests for those other scenarios.  If you have a specific scenario you believe is not tested, I will be happy to either find a test that demonstrates it is fixed, or add a test.  In the absence of a specific suggestion, your request is not actionable.

---
In reply to: [326428679](https://github.com/dotnet/roslyn/pull/38750#discussion_r326428679) [](ancestors = 326428679,326408524,326337342)

Reply:
The title is intended to convey that we are fixing **an** error scenario, not all error scenarios.

---
In reply to: [326801296](https://github.com/dotnet/roslyn/pull/38750#discussion_r326801296) [](ancestors = 326801296,326428679,326408524,326337342)